### ISSUE TITLE
page offset feature request

### ISSUE THREAD

**Issue by Quantra** [box/box-content-preview#794]
**Describe the solution you'd like**
Apply an integer page offset to entire documents via file options.  Offset is subtracted from page numbers shown to user and added to those input by user.  Used to align page numbers with printed page numbers in docs which have title, contents or other pages before page 1.

**Describe alternatives you've considered**
Have tried overloading functions to achieve this, which was possible with pre release versions of box-content-preview.

**Additional context**
I have written code that does this and am opening this issue as per CONTRIBUTING.md (because I wrote code first then read CONTRIBUTING.md) but I have not yet made a pull request.  You can find my fork/branch here:  https://github.com/Quantra/box-content-preview/tree/add-page-offset

Please let me know your thoughts =]

**Comment by jeremypress** [comment]
Hi @Quantra,
Thanks for the request! I think I managed to replicate the functionality you're going for by overriding the setPage method (check out this [codepen](https://codepen.io/jpress/pen/jxXwQj). We should be able to avoid a PR here so there is less option specific code to maintain in the library. Definitely a cool use case, but we haven't received any other requests for something like this in the past.

Let me know if the logic in the codepen works with what you're trying to solve.

**Comment by Quantra** [comment]
Hi @jeremypress 
Thanks for getting back to me.  I've tested the setPage method you provide with my own documents (books). It works (with positive offsets) in the basic principle but when pressing next/prev page controls it will skip `offset` pages and the page numbers shown to the user in the page controls don't align with those in the book or those they enter.

I used a similar technique to achieve this pre release but with pageNumInputBlurHandler.  I also added code to update the page number shown in controls.  The behaviour I describe above is what I encountered when I updated and what I have been trying to resolve.  After some time overriding various methods and chasing my tail I felt a more direct approach was required and forked.

Completely understand if you don't want to PR a feature only requested by 1.  I have my fork I'm self hosting in production, my users are happy =]
